components/brave_sync/brave_profile_sync_service_impl.cc

```diff
@@ -171,6 +171,19 @@ BraveProfileSyncServiceImpl::BraveProfileSyncServiceImpl(Profile* profile,
   brave_sync_prefs_ =
       std::make_unique<prefs::Prefs>(sync_client_->GetPrefService());
 
+  auto devices = brave_sync_prefs_->GetSyncDevices();
+  if (devices->size() < 2 &&
+      (brave_sync_prefs_->GetSyncEnabled() ||
+       !brave_sync_prefs_->GetSeed().empty() ||
+       !brave_sync_prefs_->GetThisDeviceName().empty())) {
+    // We want to reset the chain because it was not fully created
+    // We cannot make it here directly, because some sync classes may not be
+    // constructed yet
+    reseting_ = true;
```

Comment:
>Is this to trigger ResetSyncInternal() by ForceCompleteReset()?
yes

but if we don't reset devices list, UI page will never trigger BraveProfileSyncServiceImpl::OnSetupSyncHaveCode because device name is not empty https://github.com/brave/brave-core/blob/master/components/brave_sync/ui/components/modals/enterSyncCode.tsx#L92

so instead of clearing all prefs it would be enough co clean only the current device name, but I decided to clean all non-actual data